best-practices.md

```diff
@@ -23,13 +23,6 @@ All rules below should be followed by contributors to this repo. Contributors sh
     - It is allowed to have one resource allocated dynamically, and another allocated statically in the same task.
     - It is *not* allowed to have a resource which can be allocated *either* statically or dynamically.
       - e.g. `memory_gb` and `modify_memory_gb` cannot be present in the same task.
-- All tasks and workflows should have a `max_retries` input.
-  - This should be defaulted to `1` for nearly all tasks
-  - Some tasks are particularly error prone and can have a higher default `max_retries`
-  - **rule specific to workflows:** `max_retries` should be an optional `Int?`
-    - This allows each task to have it's own specific default `max_retries`
-      - If a user does not supply `max_retries`, those task level defaults will get used
-      - If a user does supply `max_retries`, it should override the default for *every* task called
 - multi-core tasks should *always* follow the conventions laid out in the `use_all_cores_task` example (see `template/task-templates.wdl`)
   - this is catering to cloud users, who may be allocated a machine with more cores than are specified by the `ncpu` parameter
```

Comment:
I wonder if we should note that `ncpu` will be deprecated when a task body can access `runtime` fields.

Reply:
I don't think it hurts. But we _are_ planning to keep the `use_all_cores` convention regardless, correct?

Reply:
I think so, but we've only retained the `ncpu` input parameter because we can't access the runtime. Hopefully with 2.0 
```
        n_cores=~{ncpu}
        if ~{use_all_cores}; then
            n_cores=$(nproc)
        fi
```
Will become something like
```
        n_cores=~{task.runtime.cpu}
        if ~{use_all_cores}; then
            n_cores=$(nproc)
        fi
```
And we can eliminate the `input` parameter in favor of querying the `runtime` directly.